firebase_admin/dynamic_links.py

```diff
@@ -169,47 +147,96 @@ def from_json(cls, platform, event, count):
-        if not isinstance(platform, str) or platform not in self._platforms.values():
-            raise ValueError('platform {}, not recognized'.format(platform))
-        self._platform = platform
-
     @property
     def event(self):
         return self._event
 
-    @event.setter
-    def event(self, event):
-        if isinstance(event, str) and event in self._event_types.keys():
-            raise ValueError(('Raw string {} detected. Use one of the dynamic_links.EVENT_TYPES_' +
-                              ' constants, or the from_json() method.').format(event))
-        if not isinstance(event, str) or event not in self._event_types.values():
-            raise ValueError('event_type {}, not recognized'.format(event))
-        self._event = event
-
     @property
     def count(self):
         return self._count
 
-    @count.setter
-    def count(self, count):
-        if not isinstance(count, int) or isinstance(count, bool) or count < 0:
-            raise ValueError('Count: {} must be a non negative int'.format(count))
-        self._count = count
 
+class StatOptions(object):
+    def __init__(self, duration_days):
+        self.duration_days = duration_days
+
+    @property
+    def duration_days(self):
+        return self._duration_days
+
+    @duration_days.setter
+    def duration_days(self, duration_days):
+        if (isinstance(duration_days, bool)
+                or not isinstance(duration_days, int)
+                or duration_days < 1):
+            raise ValueError('duration_days must be positive integer (got {})'
+                             .format(duration_days))
+        self._duration_days = duration_days
+
+class _LinksService(object):
+    """Provides methods for the Firebase dynamic links interaction"""
 
-def _validate_url(url):
-    if not isinstance(url, six.string_types) or not url.startswith('https://'):
-        raise ValueError('Url must be a string and begin with "https://".')
+    INTERNAL_ERROR = 'internal-error'
 
-def _validate_stat_options(options):
-    if not isinstance(options, StatOptions):
-        raise ValueError('Options must be of type StatOptions.')
-    if (isinstance(options.duration_days, bool)
-            or not isinstance(options.duration_days, int)
-            or options.duration_days < 1):
-        raise ValueError('duration_days: {} must be positive int'.format(options.duration_days))
+    def __init__(self, app):
+        self._client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_LINKS_BASE_URL)
+        self._timeout = app.options.get('httpTimeout')
+        self._request_string = '{0}/linkStats?durationDays={1}'
+
+    def _format_request_string(self, short_link, options):
+        days = options.duration_days
+        # Complaints about the named second argument needed to replace "/"
+        #pylint: disable=redundant-keyword-arg
+        url_quoted = urllib.parse.quote(short_link, safe='')
```

Comment:
I think if you add the `pylint` directive to the end of this line, things should workout:

```
url_quoted = urllib.parse.quote(short_link, safe='') #pylint: disable=redundant-keyword-arg
```